sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayServiceConfigurationReader.java

```diff
@@ -42,118 +24,28 @@
-    private final String authKeyResourceToken;
-    private HttpClient httpClient;
+    private GlobalEndpointManager globalEndpointManager;
 
-    public GatewayServiceConfigurationReader(URI serviceEndpoint, boolean hasResourceToken, String resourceToken,
-            ConnectionPolicy connectionPolicy, BaseAuthorizationTokenProvider baseAuthorizationTokenProvider,
-            HttpClient httpClient) {
+    public GatewayServiceConfigurationReader(URI serviceEndpoint, GlobalEndpointManager globalEndpointManager) {
         this.serviceEndpoint = serviceEndpoint;
-        this.baseAuthorizationTokenProvider = baseAuthorizationTokenProvider;
-        this.hasAuthKeyResourceToken = hasResourceToken;
-        this.authKeyResourceToken = resourceToken;
-        this.connectionPolicy = connectionPolicy;
-        this.httpClient = httpClient;
+        this.globalEndpointManager = globalEndpointManager;
+        this.globalEndpointManager.getDatabaseAccountFromCache(this.serviceEndpoint).block();
     }
 
     public ReplicationPolicy getUserReplicationPolicy() {
-        this.throwIfNotInitialized();
-        return this.userReplicationPolicy;
+        return BridgeInternal.getReplicationPolicy(this.globalEndpointManager.getLatestDatabaseAccount());
     }
 
     public ReplicationPolicy getSystemReplicationPolicy() {
-        this.throwIfNotInitialized();
-        return this.systemReplicationPolicy;
-    }
-
-    public boolean enableAuthorization() {
-        return true;
+        return BridgeInternal.getSystemReplicationPolicy(this.globalEndpointManager.getLatestDatabaseAccount());
     }
 
     public ConsistencyLevel getDefaultConsistencyLevel() {
-        this.throwIfNotInitialized();
-        return this.consistencyLevel;
-    }
-
-    public void setDefaultConsistencyLevel(ConsistencyLevel value) {
-        this.throwIfNotInitialized();
-        this.consistencyLevel = value;
+        return BridgeInternal.getConsistencyPolicy(this.globalEndpointManager.getLatestDatabaseAccount()).getDefaultConsistencyLevel();
     }
 
     public Map<String, Object> getQueryEngineConfiguration() {
-        this.throwIfNotInitialized();
-        return this.queryEngineConfiguration;
-    }
-
-    private Mono<DatabaseAccount> getDatabaseAccountAsync(URI serviceEndpoint) {
-
-        HttpHeaders httpHeaders = new HttpHeaders();
-        httpHeaders.set(HttpConstants.HttpHeaders.VERSION, HttpConstants.Versions.CURRENT_VERSION);
-
-        UserAgentContainer userAgentContainer = new UserAgentContainer();
-        String userAgentSuffix = this.connectionPolicy.getUserAgentSuffix();
-        if (userAgentSuffix != null && userAgentSuffix.length() > 0) {
-            userAgentContainer.setSuffix(userAgentSuffix);
-        }
-
-        httpHeaders.set(HttpConstants.HttpHeaders.USER_AGENT, userAgentContainer.getUserAgent());
-        httpHeaders.set(HttpConstants.HttpHeaders.API_TYPE, Constants.Properties.SQL_API_TYPE);
```

Comment:
as this `httpHeaders.set(HttpConstants.HttpHeaders.API_TYPE, Constants.Properties.SQL_API_TYPE);` is removed, we are not passing this header to the service in the initialization time. 

* what does .Net do?
* Is that ok not to set SQL_API_TYPE? what happens on non sql type account?

Reply:
Checked with Jake , for .NET code is there to send apiType on all gateway calls , but currently it is taking ApiType.None, which is is filtered out when call goes to gateway.
Now for java we have one place for all gateway calls ,we should include that in common place . We can track this item separately

Reply:
thanks